pennylane/_qubit_device.py

```diff
@@ -98,7 +103,7 @@ def _scatter(indices, array, new_dimensions):
 
     observables = {"PauliX", "PauliY", "PauliZ", "Hadamard", "Hermitian", "Identity"}
 
-    def __init__(self, wires=1, shots=1000, analytic=True):
+    def __init__(self, wires=1, shots=1000, analytic=True, cache=0):
```

Comment:
> why caching was on the Tape/QNode side and why we didn't just directly place it in QubitDevice.execute()

Was the reason that there are devices which do not inherit from `QubitDevice`? That is, does this change now disable caching for devices like `default.tensor`, CV based devices, etc.?